File: Week 2/Problem Set 2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    //Ask user for an input text.
    string text = get_string("Text: ");

    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    //Coleman-Liau index
    float index = (0.0588 * letters / words * 100) - (0.296 * sentences / words * 100) - 15.8;

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index <= 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
    }
}

//Counts the number of letter in a given text.
int count_letters(string text)
{
    int letters = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        //Letters counter will increase only if the the element is uppercase or lowercase letter.
        if (islower(text[i]) || isupper(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

//Counts the number of words in a given text.
int count_words(string text)
{
    int words = 1;
    for (int j = 0; j < strlen(text); j++)
    {
        //Assume that the sentence does not start or end with a space and there are not double spaces.
        if (text[j] == 32)
        {
            words++;
        }
    }
    return words;
}

//Counts the number of sentences in a given text.
int count_sentences(string text)
{
    int sentences = 0;
    for (int k = 0; k < strlen(text); k++)
    {
        //Assume that the sentence is a sequence of characters ending with period, exclamation point or question mark.
        //Does not take in count perion inside of the sentence like Mr.
        if (text[k] == 33 || text[k] == 46 || text[k] == 63)
        {
            sentences++;
        }
    }
    return sentences;
}

File: Week 6/Problem Set 6/sentimental-readability/readability.py
def main():
    # Ask user for input text
    text = input("Text: ")

    # L is the average number of letter per 100 words in the text
    L = 100 * count_letters(text) / count_words(text)

    # S is the average number of sentences per 100 words in the text
    S = 100 * count_sentences(text) / count_words(text)

    # Coleman-Liau index
    index = 0.0588 * L - 0.296 * S - 15.8

    print(grade(index))


# Assume that any sequence of characters separated by spaces should count as a word
def count_words(text):
    '''Return number of words in a given text'''
    words = 1
    for i in text:
        if i == " ":
            words += 1
    return words


#  Assume that a letter is any lowercase character from a to z or any uppercase character from A to Z
def count_letters(text):
    '''Return number of letters in a given text'''
    alphabet = "abcdefghijklmnopqrstvuwxyz"
    letters = 0
    for i in text:
        if i.lower() in alphabet:
            letters += 1
    return letters


# Assume that any occurrence of a period, exclamation point, or question mark indicates the end of a sentence
def count_sentences(text):
    '''Return number of sentences in a given text'''
    sentences = 0
    for i in text:
        if i == "." or i == "!" or i == "?":
            sentences += 1
    return sentences


def grade(index):
    '''Return the grade level of the given text based on its difficulty'''
    if index >= 16:
        return "Grade 16+"
    elif index < 1:
        return "Before Grade 1"
    else:
        return f"Grade {round(index)}"


if __name__ == "__main__":
    main()